draft-ietf-quic-transport.md

```diff
@@ -1639,245 +1639,190 @@ to get blocked.
 
 # Error Codes {#error-handling}
 
-This section lists all the QUIC error codes that may be used in a
-CONNECTION_CLOSE frame.  TODO: Trim list and group errors for readabiity.
+Error codes are 32 bits long, with the first two bits indicating the source of
+the error code:
 
-TODO: Discuss error handling beyond just listing error codes.
+0x0000-0x3FFF:
+: QUIC transport error codes, including packet protection errors.  Applicable to
+  all uses of QUIC.
 
-* 0x01: QUIC_INTERNAL_ERROR.  (Connection has reached an invalid state.)
+0x4000-0x7FFF:
+: Cryptographic error codes.  Defined by the crypto handshake protocol in use.
 
-* 0x02: QUIC_STREAM_DATA_AFTER_TERMINATION.  (There were data frames after the a
-  fin or reset.)
+0x8000-0xAFFF:
+: Loss-detection and congestion control error codes.  Defined by the
+  loss-detection and congestion control algorithm currently in use.
```

Comment:
Can you explain why?  Is it that congestion control is not supposed to generate errors?  Or do you expect to be able to take whatever error codes are needed from the core transport given that the two docs are more tightly coupled than the others?